test/parallel_api/algorithm/alg.modifying.operations/transform_binary.pass.cpp

```diff
@@ -47,9 +58,9 @@ check_and_reset(InputIterator1 first1, InputIterator1 last1, InputIterator2 firs
         // check
         Out expected = Out(1.5) + *first1 - *first2;
         Out actual = *out_first;
-        if (::std::is_floating_point_v<Out>)
+        if constexpr (::std::is_floating_point_v<Out>)
```

Comment:
I looked  on that one more time... 
And I think that more right way to add a proper (floating point) specialization(s) for  `expect_equal_val`, see https://github.com/oneapi-src/oneDPL/blob/main/test/support/utils.h#L102C1-L102C17
If we do it we will not need any `if constexpr (::std::is_floating_point_v<Out>) ` here.